src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -289,23 +289,47 @@ private void CloseStreamFromDispose(bool disposing)
                 base.DisposeAsync() :
                 DisposeAsyncCore();
 
-        private ValueTask DisposeAsyncCore()
+        private async ValueTask DisposeAsyncCore()
         {
-            // Same logic as in Dispose().
+            // Same logic as in StreamWriter.DisposeAsync().
             Debug.Assert(GetType() == typeof(StreamReader));
             try
             {
                 if (!_disposed)
                 {
+                    ThrowIfDisposed();
                     CheckAsyncTaskInProgress();
                 }
             }
             finally
             {
-                CloseStreamFromDispose(disposing: true);
+                await CloseStreamFromDisposeAsync(disposing: true).ConfigureAwait(false);
             }
             GC.SuppressFinalize(this);
-            return ValueTask.CompletedTask;
+        }
+        private async ValueTask CloseStreamFromDisposeAsync(bool disposing)
```

Comment:
This parameter shouldn't be necessary.

Reply:
yeah I will remove this on Writer and Reader